src/modules/comments.js

```diff
@@ -0,0 +1,45 @@
+import commentCounter from './commentCounter.js';
+
+const header = document.querySelector('.comments-counter');
+
+const baseUrl = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/iHlfVWj35yMMl9mYGaLB/';
+
+const fetchComment = async (id) => {
+  header.innerHTML = 'comments (0)';
+
+  await fetch(`${baseUrl}comments?item_id=${id}`)
+    .then((response) => response.json())
+    .then((json) => {
+      let data = json;
+      if (json?.error) {
+        data = [];
+      }
+
+      commentCounter(data, header);
+      const commentList = document.querySelector('.comments-generator');
+      commentList.innerHTML = '';
+      data.map((comment) => { // eslint-disable-line
+        const li = document.createElement('li');
+        li.innerHTML = `<span class="date">${comment.creation_date}</span><span class="date">${comment.username}:</span><span class="date">${comment.comment}</span>`;
+        commentList.appendChild(li);
+      });
+    });
+};
```

Comment:
- [ ] [OPTIONAL] Well done for using `async` and `await` for your API calls. However, you could do it better without the `.then`. I will show you some examples you could try out. You can [read more about async and await here](https://javascript.info/async-await);

```
const fetchComment = async (id) => {
  header.innerHTML = 'comments (0)';
// Notice how I refactor this section to make your code use async and await correctly.
  const response = await fetch(`${baseUrl}comments?item_id=${id}`);
  const data = await response.json();

  if (data?.error) {
    data = [];
  }

  commentCounter(data, header);
  const commentList = document.querySelector('.comments-generator');
  commentList.innerHTML = '';
  data.map((comment) => { // eslint-disable-line
    const li = document.createElement('li');
    li.innerHTML = `<span class="date">${comment.creation_date}</span><span class="date">${comment.username}:</span><span class="date">${comment.comment}</span>`;
    commentList.appendChild(li);
  });
};
```